spec/models/manageiq/providers/amazon/cloud_manager/event_catcher/stream_spec.rb

```diff
@@ -1,24 +1,126 @@
+            expect(subject).to receive(:sqs_create_queue).and_call_original
+            expect(subject).to receive(:subscribe_topic_to_queue).and_call_original
+            expect(subject.send(:find_or_create_queue)).to eq(queue_url)
+          end
+        end
+      end
 
-    @ems_stream = described_class.new(@ems.authentication_userid,
-                                      @ems.authentication_password,
-                                      @ems.provider_region,
-                                      @ems.guid)
+      context "and no topic present on aws" do
+        it "raises ProviderUnreachable" do
+          stubbed_responses = {
+            :sqs => {
+              :get_queue_url => 'NonExistentQueue'
+            },
+            :sns => {
+              :list_topics => {
+                :topics => [{:topic_arn => "arn:aws:sns:region:account-id:not-the-right-topic"}]
+              }
+            }
+          }
+          with_aws_stubbed(stubbed_responses) do
+            expect { subject.send(:find_or_create_queue) }.to raise_exception(described_class::ProviderUnreachable)
+          end
+        end
+      end
+    end
   end
 
-  context "#each_batch" do
-    it "raises ProviderUnreachable on non existing queue" do
-      expect(@ems_stream).to receive(:sqs).and_raise(AWS::SQS::Errors::NonExistentQueue)
-      allow(@ems_stream).to receive_messages(:sns => double(:topics => double(:detect => nil)))
+  context "#parse_event" do
+    let(:message) do
+      body = File.read(File.join(File.dirname(__FILE__), "sqs_message.json"))
+      Aws::SQS::Types::Message.new(:body => body, :message_id => 1)
+    end
+
+    it "parses a SNS Message" do
+      expect(subject.send(:parse_event, message)).to include('messageId'   => 1,
+                                                             "messageType" => "ConfigurationItemChangeNotification",
+                                                             "eventType"   => "AWS_EC2_Instance_UPDATE")
+    end
+  end
 
-      @ems_stream.start
-      expect do
-        @ems_stream.each_batch
-      end.to raise_error(described_class::ProviderUnreachable)
+  context "#poll" do
+    it "yields an event" do
+      message_body = File.read(File.join(File.dirname(__FILE__), "sqs_message.json"))
+      stubbed_responses = {
+        :sqs => {
+          :receive_message => [
+            {
+              :messages => [
+                {
+                  :body           => message_body,
+                  :receipt_handle => 'receipt_handle',
+                  :message_id     => 'id'
+                }
+              ]
+            },
+            # second message raises an exception
+            "Aws::SQS::Errors::ServiceError"
+          ]
+        }
+      }
+      with_aws_stubbed(stubbed_responses) do
+        allow(subject).to receive(:find_or_create_queue).and_return(queue_url)
+        allow(subject).to receive(:parse_event).and_return(message_body)
+        expect(subject)
```

Comment:
this is slipped in :(
I can remove it and push again and wait for CI to get green again